ibtk/include/ibtk/HierarchyIntegrator.h

```diff
@@ -220,9 +220,39 @@ class HierarchyIntegrator : public SAMRAI::mesh::StandardTagAndInitStrategy<NDIM
      * GriddingAlgorithm::regidAllFinerLevels() to regrid the patch hierarchy.
      * Subclasses can control the method used to regrid the patch hierarchy by
      * overriding this public virtual member function.
+     *
+     * Before regridding, this method calls regridHierarchyBeginSpecialized()
+     * on the current integrator and all child integrators.
+     *
+     * After regridding and (optionally) checking the new grid volume, this
+     * method calls regridHierarchyEndSpecialized() on the current integrator
+     * and all child integrators. It then calls the following methods (and,
+     * therefore, the specialized methods on the current and all child
+     * integrators) in the following order:
+     *
+     * 1. initializeCompositeHierarchyData
+     * 2. synchronizeHierarchyData
+     *
+     * @warning This class assumes, but does not enforce, that this method is
+     * only called on the parent integrator. A future release of IBAMR will
+     * enforce this assumption.
      */
     virtual void regridHierarchy();
 
+    /*!
+     * Perform any necessary work relevant to data owned by the current
+     * integrator prior to regridding (e.g., calculating divergences). An
+     * empty default implementation is provided.
+     */
+    virtual void regridHierarchyBeginSpecialized();
```

Comment:
I think that I usually make these kinds of methods `protected`, since they are not intended to be part of the public interface. Can we do that here?